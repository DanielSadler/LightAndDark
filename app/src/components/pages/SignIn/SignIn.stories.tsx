import { storiesOf } from "@storybook/react-native";
import React from "react";

import { SignIn } from "./SignIn";

storiesOf("Pages/SignIn", module).add("default", (args) => (
  <SignIn {...args} />
));
