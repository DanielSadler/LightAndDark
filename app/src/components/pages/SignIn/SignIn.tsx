import { useFormik } from "formik";
import { useContext } from "react";
import { Image, View } from "react-native";

import { AuthContext } from "../../../context";
import { Button, TextInput } from "../../atoms";

interface Props {}

export const SignIn = (props: Props) => {
  const { signIn } = useContext(AuthContext);

  const { values, handleChange, handleBlur, handleSubmit } = useFormik({
    initialValues: {
      email: "",
      password: "",
    },
    onSubmit: signIn,
  });
  return (
    <View className="w-full h-full items-center justify-around">
      <Image
        source={require("../../../../assets/icon.png")}
        className="h-24 w-24 pt-6"
      />
      <View className="w-full items-center">
        <TextInput
          label="Email Address"
          keyboardType="email-address"
          className="w-11/12 pb-3"
          onChangeText={handleChange("email")}
          onBlur={handleBlur("email")}
          value={values.email}
        />
        <TextInput
          label="Password"
          secureTextEntry
          className="w-11/12 pb-6"
          onChangeText={handleChange("password")}
          onBlur={handleBlur("password")}
          value={values.password}
        />
        <Button
          title="Sign In"
          className="w-11/12"
          onPress={() => handleSubmit()}
        />
      </View>
      <View />
    </View>
  );
};
