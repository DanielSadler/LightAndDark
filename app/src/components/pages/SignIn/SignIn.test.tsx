import * as React from "react";
import * as renderer from "react-test-renderer";

import { SignIn } from "./SignIn";

describe("SignIn", () => {
  it("should match a snapshot", () => {
    const tree = renderer.create(<SignIn />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
