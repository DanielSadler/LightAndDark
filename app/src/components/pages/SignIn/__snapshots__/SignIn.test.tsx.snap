// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match a snapshot 1`] = `
<View
  style={
    [
      {
        "width": "100%",
      },
      {
        "height": "100%",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "space-around",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      [
        {
          "height": 96,
        },
        {
          "width": 96,
        },
        {
          "paddingTop": 24,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "width": "100%",
        },
        {
          "alignItems": "center",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "width": "91.666667%",
          },
          {
            "paddingBottom": 12,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
            {
              "fontWeight": "500",
            },
            {
              "color": "#374151",
            },
            {
              "marginBottom": 4,
              "marginTop": 4,
            },
          ]
        }
      >
        Email Address
      </Text>
      <TextInput
        keyboardType="email-address"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        style={
          [
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            {
              "borderBottomWidth": 1,
              "borderLeftWidth": 1,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            {
              "borderBottomColor": "#d1d5db",
              "borderLeftColor": "#d1d5db",
              "borderRightColor": "#d1d5db",
              "borderTopColor": "#d1d5db",
            },
            {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            {
              "paddingBottom": 8,
              "paddingTop": 8,
            },
            {
              "color": "#000",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 2,
            },
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
        value=""
      />
    </View>
    <View
      style={
        [
          {
            "width": "91.666667%",
          },
          {
            "paddingBottom": 24,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
            {
              "fontWeight": "500",
            },
            {
              "color": "#374151",
            },
            {
              "marginBottom": 4,
              "marginTop": 4,
            },
          ]
        }
      >
        Password
      </Text>
      <TextInput
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={true}
        style={
          [
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            {
              "borderBottomWidth": 1,
              "borderLeftWidth": 1,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            {
              "borderBottomColor": "#d1d5db",
              "borderLeftColor": "#d1d5db",
              "borderRightColor": "#d1d5db",
              "borderTopColor": "#d1d5db",
            },
            {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            {
              "paddingBottom": 8,
              "paddingTop": 8,
            },
            {
              "color": "#000",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 2,
            },
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
        value=""
      />
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#6366f1",
          },
          {
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          {
            "marginBottom": 12,
            "marginLeft": 12,
            "marginRight": 12,
            "marginTop": 12,
          },
          {
            "width": "91.666667%",
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                {
                  "color": "#ffffff",
                },
              ]
            }
          >
            Sign In
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View />
</View>
`;
